lib/socket.c

```diff
@@ -742,6 +742,20 @@ int connect_socket_nb(socket_t *r, int type, const endpoint_t *ep) {
 	return connect_socket_retry(r);
 }
 
+int connect_socket_nb_timeout(socket_t *r, int type, const endpoint_t *ep, unsigned int timeout) {
```

Comment:
Suggestion: connect_socket_nb() could now be turned into a wrapper around connect_socket_nb_timeout() passing a zero timeout to eliminate some code duplication.

Alternatively usertimeout() could simply be called after the socket has been created, explicitly right after connect_socket_nb(). AFAICT it has no effect on connection progress and so setting the option after connect() should be fine, also eliminating the code duplication.